File: tp1_2_3/python/src/personne.py
class Personne:
    def __init__(self, nom, age):
        self.nom = nom
        self.age = age

    @property
    def nom(self):
        return self.__nom

    @nom.setter
    def nom(self, value):
        if not value or not value.strip():
            raise ValueError("Le nom ne peut pas être vide")
        self.__nom = value

    @property
    def age(self):
        return self.__age

    @age.setter
    def age(self, value):
        if value < 0 or value > 100:
            raise ValueError("L'âge doit être compris entre 0 et 100")
        self.__age = value

    def afficher_details(self):
        print(f"Nom : {self.__nom}, Age : {self.__age}")

File: tp1/java/main/java/heritage/Personne.java
package heritage;

public class Personne {
    private String nom;
    private int age;

    public Personne(String nom, int age) {
        setNom(nom);
        setAge(age);
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        if (nom == null || nom.trim().isEmpty()) {
            throw new IllegalArgumentException("Le nom ne peut pas être vide");
        }
        this.nom = nom;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        if (age < 0 || age > 100) {
            throw new IllegalArgumentException("L'âge doit être compris entre 0 et 100");
        }
        this.age = age;
    }

    public void afficherDetails() {
        System.out.println("Nom : " + nom + ", Age : " + age);
    }
}
